### ISSUE TITLE
error_nil/error_is_as: IsType vs nil/error

### ISSUE THREAD

**Issue by ccoVeille** [Antonboom/testifylint#240]
By looking of code using IsType, I found bad code that are not linted yet.

https://github.com/PacktPublishing/Hands-On-Serverless-Applications-with-Go/blob/f1701bc78e92fd971246fd4bced7c7bddbf22fec/ch10/findAll/main_test.go#L17

```go
assert.IsType(t, nil, err)
```

=> assert.NoError should be considered

https://github.com/Infomaniak/kchat-webapp/blob/932ec3af126f6225f9424624dad54127ac24f7ca/server/channels/store/storetest/desktop_tokens_store.go#L36

```go
assert.IsType(t, &store.ErrNotFound{}, err)
```

=> ErrorAs should be considered
